src/daft-plan/src/builder.rs

```diff
@@ -21,23 +23,8 @@ pub struct LogicalPlanBuilder {
 }
 
 impl LogicalPlanBuilder {
-    // Create a new LogicalPlanBuilder for a Source node.
-    pub fn from_source(source: ops::Source) -> Self {
-        Self {
-            plan: LogicalPlan::Source(source).into(),
-        }
-    }
-
-    pub fn from_filter(filter: ops::Filter) -> Self {
-        Self {
-            plan: LogicalPlan::Filter(filter).into(),
-        }
-    }
-
-    pub fn from_limit(limit: ops::Limit) -> Self {
-        Self {
-            plan: LogicalPlan::Limit(limit).into(),
-        }
+    pub fn new(plan: Arc<LogicalPlan>) -> Self {
+        Self { plan }
```

Comment:
Nice! I had previously done something similar via implementing the `From<LogicalPlan>` trait, but this works nicely as well.